hw/nfc-nci.c

```diff
@@ -745,8 +751,10 @@ init_process_oid_rf_deactivate_cmd(const union nci_packet* cmd,
         data->type = payload->type;
         data->reason = NCI_RF_DEACT_DH_REQUEST;
 
-        nfc_delivery_cb_setup(cb, NTFN_BUF, data,
+        nfc_delivery_cb_setup(nfc, cb, NTFN_BUF, data,
             nfc_delivery_deactivate_cmd_cb);
+    } else {
+        nfc_rf_state_deactivate_transition(payload, nfc);
```

Comment:
I'd like this to be a bit closer to the spec [NCI], Sec 5.2.2 and others, which says

'If the DH sends RF_DEACTIVATE_CMD, the NFCC SHALL ignore the Deactivation Type
parameter, stop the RF Discovery Process, and send RF_DEACTIVATE_RSP. The state will then change to RFST_IDLE.'

We *first* need to send the RSP and *then* change to RFST_IDLE.

Please change the code to

  - set up the delivery callback unconditionally,
  - store send_ntf in the callback data,
  - and decide in |nfc_delivery_deactivate_cmd_cb| whether to send the NTF or not.